runtime/vm/threadpark.cpp

```diff
@@ -62,6 +66,19 @@ threadParkImpl(J9VMThread *vmThread, BOOLEAN timeoutIsEpochRelative, I_64 timeou
 		}
 		thrstate |= J9_PUBLIC_FLAGS_THREAD_TIMED;
 	}
+#if defined(J9VM_OPT_CRIU_SUPPORT)
+	else {
+		/* timeoutIsEpochRelative is false and timeout = 0 */
+		if (J9_IS_SINGLE_THREAD_MODE(vm)
+			&& VM_VMHelpers::threadCanRunJavaCode(vmThread)
```

Comment:
Instead of this we can check if its the checkpoint thread. `vm->checkpointState.checkpointThread`